agent/queryparser/helpers.go

```diff
@@ -0,0 +1,174 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package queryparser provides functionality for queries parsing.
+package queryparser
+
+import (
+	"bufio"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+var (
+	spaceRegexp *regexp.Regexp
+	spaceOnce   sync.Once
+	errSpace    error
+
+	multilineRegexp *regexp.Regexp
+	multilineOnce   sync.Once
+	errMultiline    error
+)
+
+func parseMySQLComments(q string) (map[string]bool, error) {
+	prepareMultilineRegexp()
+	if errMultiline != nil {
+		return nil, errMultiline
+	}
+
+	// comments using comment as a key to avoid duplicates
+	comments := make(map[string]bool)
+	for _, v := range multilineRegexp.FindAllStringSubmatch(q, -1) {
+		if len(v) < 2 {
+			continue
+		}
+
+		value := removeFormatting(v[1])
+
+		var err error
+		value, err = removeSpaces(value)
+		if err != nil {
+			return nil, err
+		}
+
+		comments[value] = true
+	}
+
+	hashComments, err := parseSinglelineComments(q, "#")
+	if err != nil {
+		return nil, err
+	}
+	for c := range hashComments {
+		comments[c] = true
+	}
+
+	dashComments, err := parseSinglelineComments(q, "--")
+	if err != nil {
+		return nil, err
+	}
+	for c := range dashComments {
+		comments[c] = true
+	}
+
+	return comments, nil
+}
+
+func parsePostgreSQLComments(q string) (map[string]bool, error) {
+	prepareMultilineRegexp()
+	if errMultiline != nil {
+		return nil, errMultiline
+	}
+
+	// comments using comment as a key to avoid duplicates
+	comments := make(map[string]bool)
+	for _, v := range multilineRegexp.FindAllStringSubmatch(q, -1) {
+		if len(v) < 2 {
+			continue
+		}
+
+		value := removeFormatting(v[1])
+
+		var err error
+		value, err = removeSpaces(strings.ReplaceAll(value, "*", ""))
+		if err != nil {
+			return nil, err
+		}
+
+		comments[value] = true
+	}
+
+	dashComments, err := parseSinglelineComments(q, "--")
+	if err != nil {
+		return nil, err
+	}
+	for c := range dashComments {
+		comments[c] = true
+	}
+
+	return comments, nil
+}
+
+func parseSinglelineComments(q, startChar string) (map[string]bool, error) {
+	r, err := regexp.Compile(fmt.Sprintf("%s.*", startChar))
+	if err != nil {
+		return nil, err
+	}
+
+	// comments using comment as a key to avoid duplicates
+	comments := make(map[string]bool)
+	lines, err := stringToLines(q)
+	if err != nil {
+		return nil, err
+	}
+	for _, l := range lines {
+		for _, v := range r.FindStringSubmatch(l) {
+			comments[strings.TrimLeft(v, fmt.Sprintf("%s ", startChar))] = true
+		}
+	}
+
+	return comments, nil
+}
+
+func prepareMultilineRegexp() error {
+	// to compile regexp only once
+	multilineOnce.Do(func() {
+		multilineRegexp, errMultiline = regexp.Compile(`(?s)\/\*(.*?)\*\/`)
+	})
+	if errMultiline != nil {
+		return errMultiline
+	}
+
+	return nil
+}
```

Comment:
What about calling this in `init()` for the package?

Reply:
I dont like init because it no so transparent and maybe user will not use action at all. So in case of init it will be triggered for no reason. I am suggesting to have it like that.